packages/venia-ui/lib/components/CheckoutPage/ShippingMethod/shippingMethod.js

```diff
@@ -1,46 +1,90 @@
 
+import { mergeClasses } from '../../../classify';
+
+import Done from './done';
+import Editing from './editing';
 import defaultClasses from './shippingMethod.css';
 
+import {
+    GET_SHIPPING_METHODS,
+    GET_SELECTED_SHIPPING_METHOD,
+    SET_SHIPPING_METHOD
+} from './shippingMethod.gql';
+
 const ShippingMethod = props => {
     const { onSave } = props;
 
-    // TODO: Replace "doneEditing" with a query for existing data.
-    const [doneEditing, setDoneEditing] = useState(false);
-    const handleClick = useCallback(() => {
-        setDoneEditing(true);
-        onSave();
-    }, [onSave]);
-
-    const className = doneEditing
-        ? defaultClasses.container
-        : defaultClasses.container_edit_mode;
-
-    /**
-     * TODO
-     *
-     * Change this to reflect diff UI in diff mode.
-     */
-    const shippingMethod = doneEditing ? (
-        <div>In Read Only Mode</div>
-    ) : (
-        <div>In Edit Mode</div>
-    );
-    return (
-        <div className={className}>
-            <div>Shipping Method Will be handled in PWA-179</div>
-            <div className={defaultClasses.text_content}>{shippingMethod}</div>
-            {!doneEditing ? (
-                <div className={defaultClasses.proceed_button_container}>
-                    <Button onClick={handleClick} priority="normal">
-                        {'Continue to Payment Information'}
-                    </Button>
-                </div>
-            ) : null}
-        </div>
+    const talonProps = useShippingMethod({
+        onSave,
+        queries: {
+            getShippingMethods: GET_SHIPPING_METHODS,
+            getSelectedShippingMethod: GET_SELECTED_SHIPPING_METHOD
+        },
+        mutations: {
+            setShippingMethod: SET_SHIPPING_METHOD
+        }
+    });
+
+    const {
+        displayState,
+        handleSubmit,
+        hasShippingMethods,
+        isLoadingShippingMethods,
+        isLoadingSelectedShippingMethod,
+        selectedShippingMethod,
+        shippingMethods,
+        showEditMode
+    } = talonProps;
+
+    const classes = mergeClasses(defaultClasses, props.classes);
+
+    const contentsMap = useMemo(
+        () =>
+            new Map()
+                .set(
+                    displayStates.EDITING,
+                    <Editing
+                        handleSubmit={handleSubmit}
+                        hasShippingMethods={hasShippingMethods}
+                        isLoading={
+                            isLoadingSelectedShippingMethod ||
+                            isLoadingShippingMethods
+                        }
+                        selectedShippingMethod={selectedShippingMethod}
+                        shippingMethods={shippingMethods}
+                    />
+                )
+                .set(
+                    displayStates.DONE,
+                    <Done
+                        isLoading={isLoadingSelectedShippingMethod}
```

Comment:
Would this ever get used? The only time you'll render this state is when you KNOW that you have a selected shipping method.

Reply:
That ... is an excellent point 😄 

Resolved by https://github.com/magento/pwa-studio/pull/2280/commits/377917ddd5d9fb8a0f7cfdec49bd36356389912e.